tests/behaviour/util.rs

```diff
@@ -19,8 +19,166 @@
+        _ => unreachable!(),
+    }
+}
+
+async fn key_values_equal(context: &Context, expected_label_and_value: &str, answer: &Concept) -> bool {
+    let identifiers: Vec<&str> = expected_label_and_value.splitn(2, ":").collect();
+    assert_eq!(identifiers.len(), 2, "Unexpected table cell format: {expected_label_and_value}.");
+
+    let res = match answer {
+        Concept::Entity(entity) => {
+            async { entity.get_has(context.transaction(), vec![], vec![Annotation::Key]) }
+                .and_then(|stream| async { stream.try_collect::<Vec<_>>().await })
+                .await
+        }
+        Concept::Attribute(attr) => {
+            async { attr.get_has(context.transaction(), vec![], vec![Annotation::Key]) }
+                .and_then(|stream| async { stream.try_collect::<Vec<_>>().await })
+                .await
+        }
+        Concept::Relation(rel) => {
+            async { rel.get_has(context.transaction(), vec![], vec![Annotation::Key]) }
+                .and_then(|stream| async { stream.try_collect::<Vec<_>>().await })
+                .await
+        }
+        _ => unreachable!("Unexpected Concept type: {answer:?}"),
+    };
+    match res {
+        Ok(_) => {
+            let equals = res
+                .unwrap()
+                .into_iter()
+                .filter(|attr| attr.type_.label == identifiers[0])
+                .collect::<Vec<_>>()
+                .first()
+                .and_then(|attr| Some(value_equals_str(&attr.value, identifiers[1])));
+            match equals {
+                Some(val) => val,
+                None => false,
+            }
+        }
+        Err(_) => false,
+    }
+}
+
+fn labels_equal(expected_label: &str, answer: &Concept) -> bool {
+    let mut binding = String::new();
+    let label = match answer {
+        Concept::EntityType(type_) => &type_.label,
+        Concept::RoleType(RoleType { label, .. }) => {
+            binding = format!("{label}");
+            &binding
+        }
+        Concept::Entity(Entity { type_: EntityType { label, .. }, .. }) => label,
+        Concept::Relation(Relation { type_: RelationType { label, .. }, .. }) => label,
+        Concept::RootThingType(_) => {
+            binding = String::from("thing");
+            &binding
+        }
+        Concept::RelationType(RelationType { label, .. }) => label,
+        Concept::AttributeType(AttributeType { label, .. }) => label,
+        Concept::Attribute(Attribute { type_: AttributeType { label, .. }, .. }) => label,
+    };
+    label == expected_label
+}
+
+fn values_equal(expected_label_and_value: &str, answer: &Concept) -> bool {
+    let identifiers: Vec<&str> = expected_label_and_value.splitn(2, ":").collect();
+    assert_eq!(identifiers.len(), 2, "Unexpected table cell format: {expected_label_and_value}.");
+    match answer {
+        Concept::Attribute(Attribute { value, .. }) => value_equals_str(value, identifiers[1]),
+        _ => false,
+    }
```

Comment:
You could use deconstruction here instead:
```suggestion
    let Concept::Attribute(Attribute { value, .. }) = answer else { return false; }
    // should probably be `unreachable!()`, because if that's not an Attribute something went horribly wrong
    value_equals_str(value, identifiers[1])
```